renderdoc/driver/gl/gl_replay.cpp

```diff
@@ -2236,103 +2236,117 @@ byte *GLReplay::GetTextureData(ResourceId tex, uint32_t arrayIdx, uint32_t mip,
     arraysize = texDetails.depth;
   }
 
-  if(params.remap && intFormat != eGL_RGBA8 && intFormat != eGL_SRGB8_ALPHA8)
+  if(params.remap)
   {
-    RDCASSERT(params.remap == eRemap_RGBA8);
+    GLenum remapFormat = eGL_RGBA8;
+    if(params.remap == eRemap_RGBA8)
+      remapFormat = eGL_RGBA8;
+    else if(params.remap == eRemap_RGBA16)
+      remapFormat = eGL_RGBA16;
+    else if(params.remap == eRemap_RGBA32)
+    {
+      if(!IsGLES || HasExt[EXT_color_buffer_float])
+        remapFormat = eGL_RGBA32F;
+      else
+        remapFormat = eGL_RGBA32UI;
```

Comment:
It could be, it might be more complicated to remap to a native depth format but I think it would give better results as it maintains more information about the original format.